### ISSUE TITLE
[Bug?]: use Server causes Problems (missing Information)

### ISSUE THREAD

**Issue by aronmal** [solidjs/solid-start#1299]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Current behavior 😯

I am getting node is null, sourceMap is null and Hydration Mismatch errors. See screenshots.

The errors are not appearing 100% but at 50% of the page loads. It is hard for me to make out what exactly does cause the errors to cause, but when I don't use the createResource in the returned JSX I am free of errors and the server does not seem to crash on the server function. The debugging errors are also barely helpful to me.

Because I am using fontawesome pro icons, discord api tokens and a database you can't really try out my repo. I already tried to strip down the page to have an example but then the errors are even more unreliably triggerd or seamingly not anymore. I already unit tested the server function and it does not have any problems running, the solid code should also follow all reactivity rules, so we are a bit out of ideas. I will try to later post the link to our repo so you can at least see the our component(s).

One of the the errors:

![Screenshot from 2024-01-20 17-45-35](https://github.com/solidjs/solid-start/assets/57275207/6d42075b-1e83-435d-ac3a-b9c7196171fd)

**EDIT:**

I just added the the sample repo which I stripped down to the bare minimum where no tokens or env vars are needed. When I follow the 'Steps to reproduce' I just added, I see the error. While stripping down my code, it was very wierd, sometimes it still worked sometimes it didn't. What I noticed, if you shouldn't see the errors, edit in `/src/routes/config/[guildId].tsx`:

```tsx

  await fetch("https://jsonplaceholder.typicode.com/todos/1")
    .then(() => console.log("fine", Date.now()))
    .catch(() => console.log("not fine (doesn't matter)", Date.now()));
```
from `Date.now()` to `new Date()` (while the dev server is running) like so:
```tsx
  await fetch("https://jsonplaceholder.typicode.com/todos/1")
    .then(() => console.log("fine", new Date()))
    .catch(() => console.log("not fine (doesn't matter)", new Date()));
```

Somehow it triggered the error if it should happend to work consitently before.

AND I noticed if I would add a console log before the return I didn't had any errors, especially also in my production code, but it does not feel right. Also in `/src/routes/config/[guildId].tsx`:

```tsx
...
  createEffect(() => {
    const ref = pingableRolesRef();
    if (!ref) return;
    ref.checked = config.features.timePlanning.pingableRoles;
  });

  return (
    <Layout site="config">
      <h3 class="text-center">Configure li&apos;l Judd in</h3>
      <div>
...
```
With console.log added:
```tsx
...
  createEffect(() => {
    const ref = pingableRolesRef();
    if (!ref) return;
    ref.checked = config.features.timePlanning.pingableRoles;
  });

  console.log(payload());

  return (
    <Layout site="config">
      <h3 class="text-center">Configure li&apos;l Judd in</h3>
      <div>
...
```

### Expected behavior 🤔

Being able to use createResource to fetch information for the client (even though the server seems to be executing the resource function).

### Steps to reproduce 🕹

Steps:

1. Get sample project and start dev server:
```bash
git clone https://git.moonleay.net/Websites/liljudd-website.git
cd liljudd-website/
git checkout null-bug
pnpm i
pnpm dev
```
2. Open your browser and visit `http://localhost:3000/config/598539452343648256`
3. See the error 

### Your environment 🌎

```shellSystem:
  OS: Linux pop-os 6.5.6-76060506-generic #202310061235~1697396945~22.04~9283e32 SMP PREEMPT_DYNAMIC Sun O x86_64 x86_64 x86_64 GNU/Linux
  CPU: (8c/16t) x86_64
Binaries:
  Node: v21.1.0 - /home/aronmal/.nvm/versions/node/v21.1.0/bin/node
  Yarn: 1.22.21 - /home/aronmal/.nvm/versions/node/v21.1.0/bin/yarn
  npm: 10.2.0 - /home/aronmal/.nvm/versions/node/v21.1.0/bin/npm
  pnpm: 8.14.1 - /home/aronmal/.nvm/versions/node/v21.1.0/bin/pnpm (my preference)
npmPackages:
├── @auth/core@0.22.0
├── @auth/drizzle-adapter@0.3.16
├── @auth/solid-start@0.1.2
├── @eslint-community/eslint-utils@4.4.0 extraneous
├── @eslint-community/regexpp@4.10.0 extraneous
├── @eslint/eslintrc@2.1.4 extraneous
├── @solidjs/meta@0.29.3
├── @solidjs/router@0.10.9
├── @solidjs/start@0.4.10
├── @typescript-eslint/eslint-plugin@6.19.0
├── @typescript-eslint/scope-manager@6.19.0 extraneous
├── @typescript-eslint/type-utils@6.19.0 extraneous
├── @typescript-eslint/types@6.19.0 extraneous
├── @typescript-eslint/visitor-keys@6.19.0 extraneous
├── dotenv@16.3.2
├── drizzle-kit@0.20.13
├── drizzle-orm@0.29.3
├── drizzle-zod@0.5.1
├── eslint-config-prettier@9.1.0
├── eslint-plugin-solid@0.13.1
├── eslint-scope@7.2.2 extraneous
├── eslint@8.56.0
├── moment-timezone@0.5.44
├── openapi-fetch@0.8.2
├── openapi-typescript@6.7.4
├── pg@8.11.3
├── postgres@3.4.3
├── prettier-plugin-organize-imports@3.2.4
├── prettier@3.2.4
├── sass@1.70.0
├── solid-js@1.8.11
├── typescript@5.3.3
├── vinxi@0.1.4
└── zod@3.22.4
```

**Comment by lxsmnsyc** [comment]
@aronmal the last 4 errors are completely irrelevant to your issue.

Now for the exact issue, it comes from `Layout`

**Comment by aronmal** [comment]
At first, many thanks for the respond.

Do you have a hint for me what about the Layout.tsx is wrong? It is very simple and I would have guessed I did follow the reactive rules.

Layout.tsx:

```tsx
import { JSX, Suspense } from "solid-js";
import "../styles/Layout.scss";
import Footer from "./Footer";
import NavBar from "./NavBar";

function Layout(props: { children: JSX.Element; site: string }) {
  return (
    <>
      <NavBar />
      <div class={props.site}>
        <Suspense>{props.children}</Suspense>
      </div>
      <Footer />
    </>
  );
}

export default Layout;

```

**Comment by aronmal** [comment]
Okay, I just got the code working, I now used a fetch to an api route where I just have the payload function as export GET. Also, then I got a crash because I was using useLocation in it and it said it is not in a router component (makes sense because it is a backend api route). I removed it because I just used the location.pathname for debugging. (And I had to use the APIEvent request instead of the getRequestEvent for the frontend, also makes sense) Now everything is working as expected.

Still when I use a "use server" function I get a lot of errors, the Hydration Mismatch, the sourcemap null errors and 500 errors of the dev server form the _server (for use server function I guess).

If I am able to determine the exact issue with the use server I will reach back.

**Comment by aronmal** [comment]
Thanks @ryansolid  for having a quick look. I am terribly sorry for the this issue, but I was very confused by this issue for some time now and wasn't/still am not sure about it.

I noticed I could make the errors go away with deferStream set to true for the createResource. I was also able to stop the errors just by doing console.log(resource()) which apperently also makes the rest wait until the resource has waited successfully. So there wasn't a problem with the 'use server' after all, my guess.

Also I noticed the createResource was triggered twice, on the server and on the client (Did two fetch request to the api then, rate limiting :upside_down_face: ). But this seems to be gone in the latest versions. The same applies to the createAsync which didn't updated the state on the client side in createEffect, despite rendering it correctly, but also gone now.

It probably was me all the time, in case the deferStream is need in my usecase. The incomplete docs are a shame if it would had explained it a bit better (or am I missing the point in the solid docs about the deferStream explanation?), but I can totally understand the current situation and will continue to use use and praise solid and solid-start.

Keep up the good attitude Ryan!
